flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -288,7 +288,7 @@ protected void internalDeregisterApplication(
 	}
 
 	@Override
-	public boolean startNewWorker(ResourceProfile resourceProfile) {
+	public boolean startNewWorker(WorkerResourceSpec workerResourceSpec) {
```

Comment:
I think we should ensure that `workerResourceSpec` can be fulfilled by the actual container size the `YarnResourceManager` starts.